exasol_advanced_analytics_framework/udf_communication/local_discovery.py

```diff
@@ -0,0 +1,56 @@
+import socket
+import time
+from typing import cast
+
+from exasol_advanced_analytics_framework.udf_communication.local_discovery_socket import LocalDiscoverySocket
+from exasol_advanced_analytics_framework.udf_communication.messages import PingMessage
+from exasol_advanced_analytics_framework.udf_communication.peer_communicator import PeerCommunicator
+from exasol_advanced_analytics_framework.udf_communication.serialization import serialize_message, deserialize_message
+
+
+class LocalDiscovery:
+
+    def __init__(self,
+                 discovery_timeout_in_seconds: int,
+                 peer_communicator: PeerCommunicator,
+                 local_discovery_socket: LocalDiscoverySocket
+                 ):
+        self._local_discovery_socket = local_discovery_socket
+        self._peer_communicator = peer_communicator
+        self._discovery_timeout_in_minutes = discovery_timeout_in_seconds
+        self._discover_peers()
+
+    def _is_timeout(self, begin_time_ns: int):
+        current_time_ns = time.monotonic_ns()
+        time_difference_ns = current_time_ns - begin_time_ns
+        discovery_timeout_in_ns = self._discovery_timeout_in_minutes * 10 ** 9
```

Comment:
if you only use the timeout in ns, why do you save the timeout in the member variable in minutes?

Reply:
should be fixed with rework of the LocalDiscovery